packages/devtools_app/test/visible_screens_test.dart

```diff
@@ -0,0 +1,183 @@
+          equals([
+            // InspectorScreen,
+            // TimelineScreen,
+            // MemoryScreen,
+            // PerformanceScreen,
+            DebuggerScreen,
+            NetworkScreen,
+            LoggingScreen,
+            if (codeSizeScreenEnabled) CodeSizeScreen,
+          ]));
+    });
+
+    testWidgets('are correct for Flutter (non-web) debug app',
+        (WidgetTester tester) async {
+      setupMockValues(flutter: true);
+
+      expect(
+          visibleScreenTypes,
+          equals([
+            InspectorScreen,
+            TimelineScreen,
+            MemoryScreen,
+            PerformanceScreen,
+            DebuggerScreen,
+            NetworkScreen,
+            LoggingScreen,
+            if (codeSizeScreenEnabled) CodeSizeScreen,
+          ]));
+    });
+
+    testWidgets('are correct for Flutter (non-web) profile app',
+        (WidgetTester tester) async {
+      setupMockValues(flutter: true, profileMode: true);
+
+      expect(
+          visibleScreenTypes,
+          equals([
+            InspectorScreen,
+            TimelineScreen,
+            MemoryScreen,
+            PerformanceScreen,
+            DebuggerScreen,
+            NetworkScreen,
+            LoggingScreen,
+            if (codeSizeScreenEnabled) CodeSizeScreen,
+          ]));
+    });
+
+    testWidgets('are correct for Flutter web debug app',
+        (WidgetTester tester) async {
+      setupMockValues(flutter: true, web: true);
+
+      expect(
+          visibleScreenTypes,
+          equals([
+            InspectorScreen,
+            // TimelineScreen,
+            // MemoryScreen,
+            // PerformanceScreen,
+            DebuggerScreen,
+            NetworkScreen,
+            LoggingScreen,
+            if (codeSizeScreenEnabled) CodeSizeScreen,
+          ]));
+    });
+
+    testWidgets('are correct for Flutter web profile app',
+        (WidgetTester tester) async {
+      setupMockValues(flutter: true, web: true, profileMode: true);
+
+      expect(
+          visibleScreenTypes,
+          equals([
+            InspectorScreen,
+            // TimelineScreen,
+            // MemoryScreen,
+            // PerformanceScreen,
+            DebuggerScreen,
+            NetworkScreen,
+            LoggingScreen,
+            if (codeSizeScreenEnabled) CodeSizeScreen,
+          ]));
+    });
+
+    testWidgets('are correct when offline', (WidgetTester tester) async {
+      offlineMode = true;
+      setupMockValues(web: true); // Web apps would normally hide
+
+      expect(
+          visibleScreenTypes,
+          equals([
+            // InspectorScreen,
+            TimelineScreen, // Works offline, so appears regardless of web flag
+            // MemoryScreen,
+            PerformanceScreen, // Works offline, so appears regardless of web flag
+            DebuggerScreen,
+            NetworkScreen,
+            LoggingScreen,
+            if (codeSizeScreenEnabled) CodeSizeScreen,
```

Comment:
I believe this list should only contain TimelineScreen and PerformanceScreen, as those are the only two screens that have implemented offline support

Reply:
How do I access the offline mode? I've fixed the code/tests, but unsure how to check the app still works. Going to `/snapshot` gives a message about not being able to load a screen with ID `null`, but I can't figure out how to provide it an ID.

Reply:
the easiest way would be to record a CPU profile in the Performance page and click the "Export" button. then drag and drop that file into DevTools (this only works for the web version)

Reply:
Ah, got it! Though it seems like when doing this, the navigation/tabs aren't shown (no `tabs` are passed to `DevToolsScaffold`, so it doesn't look like this change affects anything there right now.